src/+otp/+utils/+solvers/dae43.m

```diff
@@ -0,0 +1,249 @@
+function [sol, y] = dae43(f, tspan, y0, options)
+
+n = numel(y0);
+
+h = odeget(options, 'InitialStep', []);
+reltol = odeget(options, 'RelTol', 1e-3);
+abstol = odeget(options, 'AbsTol', 1e-6);
+J = odeget(options, 'Jacobian', []);
+M = odeget(options, 'Mass', speye(numel(y0)));
+MStateDependence = odeget(options, 'MStateDependence', 'none');
+
+% We won't support state-dependent Mass, simple as that
+if strcmp(MStateDependence, 'strong')
+    error('OTP:MassStateDependent', 'Strong state dependent mass is not supported.')
+end
+
+if isempty(J)
+    J = @(t, y) jacapprox(f, t, y);
+    usesparse = false;
+else
+    usesparse = issparse(J(tspan(1), y0));
+end
+
+if isempty(M)
+    if usesparse
+        M = @(t, y) speye(numel(y0));
+    else
+        M = @(t, y) eye(numel(y0));
+    end
+elseif ~isa(M, 'function_handle')
+    M = @(t, y) M;
```

Comment:
We also need to support mass matrices that are only a function of time, e.g., `M(t)`